src/backend/lib/app-manager/app-server.ts

```diff
@@ -9,18 +29,24 @@ export class AppServer {
     /**
      * Instance of `UdpServer`.
      */
-    public serverInstance = new UdpServer();
+    public serverInstance: UdpServer;
 
     /**
      * Instance of `DualshockLikeController`.
      */
-    public controller = new DualshockLikeController(0).startWatching();
+    public controllerMaster = new ControllerMaster();
+
+    public activeController: GenericController<MotionDataWithTimestamp> | null;
```

Comment:
In general I don't like having an active controller, it just complicates things. I made DualshockLikeController class for just a problem like this - it hides what kind on controller is currently active and emits events when they need to be processed. As such, ControllerMaster should be hidden within a DualshockLikeController as a static instance or something. This way in the future it would be easy to extend app to support up to 4 active controllers at once.

Reply:
This is unfortunately quite a big request for a change in your logic and code that might require you to refactor ControllerMaster quite a bit :/

Reply:
No, I see what you mean. Going through and fixing some of the bugs, it was unnecessarily complicated because there were cases where we didn't have the controller handle and we had to handle those events separately, and looking through other controller APIs, they seem to just give out a controller handle whether or not the controller is actually active. I'm going to refactor the controller master class to just have like 12 controller handles that are already allocated.